src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -51,19 +55,41 @@ public function __construct(Version $version)
         $this->version = $version;
     }
 
+    /**
+     * @param string $locale if no locale given, return URL for available languages list
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(?string $locale = null): string
+    {
+        if ($locale) {
+            return self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/' . $locale . '/' . $locale . '.zip';
+        } else {
+            return self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/available_languages.json';
+        }
```

Comment:
Just 2 little suggestions to be a bit more DRY ;)
```suggestion
        $baseUrl = self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/';
        if ($locale) {
            return $baseUrl . $locale . '/' . $locale . '.zip';
        } else {
            return $baseUrl . 'available_languages.json';
        }
```

```suggestion
        return sprintf(
            self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/%s',
            $locale
                ? $locale . '/' . $locale . '.zip'
                : 'available_languages.json'
        );
```